pkg/scannerv4/mappers/mappers.go

```diff
@@ -671,137 +643,86 @@ func pkgFixedBy(enrichments map[string][]json.RawMessage) (map[string]string, er
 	return pkgFixedBys, nil
 }
 
-// severityValues contains severity information that can retrieved from a
-// ClairCore vulnerability report.
-type severityValues struct {
-	severity string
-	v2Vector string
-	v2Score  float32
-	v3Vector string
-	v3Score  float32
-	source   v4.VulnerabilityReport_Vulnerability_CVSS_Source
-	url      string
-}
-
-// severityAndCvssMetrics processes the severity and CVSS metrics for a given vulnerability
-// based on the updater type (e.g., RHEL, NVD, OSV). This function gathers CVSS metrics data
-// from multiple sources, calculates the preferred severity with CVSS Score from preferred source, and handles any errors encountered
-// during the process.
-// Returns:
-// - cvssMetrics: A slice of CVSS metrics collected from different sources (e.g., RHEL, NVD, OSV).
-// - preferredCVSS: Stackrox preferred CVSS metric, based on the updater type.
-// - severity: A string representing the final severity level for the vulnerability, based on the updater type.
-// - err: An error, if any occurred during the calculation of the CVSS scores.
-func severityAndCvssMetrics(ctx context.Context, vuln *claircore.Vulnerability, nvdVuln *nvdschema.CVEAPIJSON20CVEItem) ([]*v4.VulnerabilityReport_Vulnerability_CVSS, *v4.VulnerabilityReport_Vulnerability_CVSS, string, error) {
+// cvssMetrics processes the CVSS metrics and severity for a given vulnerability.
+// This function gathers CVSS metrics data from multiple sources and
+// returns a slice of CVSS metrics collected from different sources (e.g., RHEL, NVD, OSV).
+// When not empty, the first entry is the "preferred" metric.
+// An error is returned when there is a failure to collect CVSS metrics from all sources;
+// however, the returned slice of metrics will still be populated with any successfully gathered metrics.
+// It is up to the caller to ensure the returned slice is populated prior to using it.
+func cvssMetrics(ctx context.Context, vuln *claircore.Vulnerability, nvdVuln *nvdschema.CVEAPIJSON20CVEItem) ([]*v4.VulnerabilityReport_Vulnerability_CVSS, error) {
```

Comment:
At this time, we only care about the "preferred" and NVD, so I think this is ok. If we did want more, I think it can be added in the future. What do you think @daynewlee ?